### PR TITLE
[ISSUE #3621 Fix Passwords containing special characters are not supported (such as character "+")

### PR DESCRIPTION
**Pull request by syapollo**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

for #3621

## Brief changelog

encode the passowrd and password can not be null

## Verifying this change

I simulated some special characters and they can pass normally

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java [hunk 1]
```diff
@@ -41,45 +43,45 @@
  * @since 1.2.0
  */
 public class SecurityProxy {
-    
+
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java:46]: Do not change the indent .
Please use nacos code style to reformat code.

detail see soruce code `style/CodeStyle.md`

**Comment by KomachiSion** [comment]
@syapollo Please check your commit author is your github id.

Otherwise the CLA will can't pass.

You can search how to change the username and email of git by search engine.

File: client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java (hunk 1 above)
Comment (syapollo) [reply client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java]: sorry
I have modified and re-uploaded

**Comment by syapollo** [comment]
> @syapollo Please check your commit author is your github id.
> 
> Otherwise the CLA will can't pass.
> 
> You can search how to change the username and email of git by search engine.

I send mail to opensource@alibaba-inc.com today

**Comment by syapollo** [comment]
@syapollo I have other suggestions . You can close this PR and re-checkout one branch from develop branch. And do these change in new branch.

Then commit with syapollo and submit by new PR.

**Pull request closed without merging**